heat/core/operations.py

```diff
@@ -231,15 +232,8 @@ def max(x, axis=None):
     #TODO: initial : scalar, optional   
     The minimum value of an output element. Must be present to allow computation on empty slice.
     """
-    #perform sanitation:
-    axis = stride_tricks.sanitize_axis(x.shape,axis)
-    
-    if axis is not None:        
-        max_axis = x._tensor__array.max(axis, keepdim=True) 
-    else:
-        return x._tensor__array.max()
 
-    return __reduce_op(x, max_axis, mpi.reduce_op.MAX, axis)
+    return __reduce_op(x, x._tensor__array.max, mpi.reduce_op.MAX, axis)
```

Comment:
We should use the high-level torch module function there, i.e. torch.max and then inside __reduce_op do partial_op(x). This way we can pass bound and unbound functions to __reduce_op